backend/tests/buildings.test.js

```diff
@@ -2,45 +2,43 @@ const request = require('supertest');
 const app = require('../App');
 
 describe('Buildings', () => {
+  let token = null;
+
+  before((done) => {
+    request(app)
+      .post('/users/login')
+      .send({ username: 'testuser2', password: '12345678' })
+      .end((err, res) => {
+        token = res.body.token;
+        done();
+      });
+  });
+
   describe('GET /kingdom/buildings', () => {
     it('should return 200 when getting all buildings', (done) => {
       request(app)
         .get('/kingdom/buildings')
+        .set('Authorization', `bearer ${token}`)
         .expect('Content-Type', /json/)
         .expect(200, done);
-    });
+    }).timeout(5000);
   });
 
   describe('GET /kingdom/buildings/non-number', () => {
-    it('should return 400 when the building id is not a number', (done) => {
+    it('should return 403 when the building id is not a number', (done) => {
       request(app)
         .get('/kingdom/buildings/non-number')
-        .expect(400, done);
-    });
-  });
-
-  describe('GET /kingdom/buildings/-1', () => {
-    it('should return 400 when trying to get a building with id -1', (done) => {
-      request(app)
-        .get('/kingdom/buildings/-1')
-        .expect(400, done);
+        .expect(403, done);
     });
   });
 
-  describe('GET /kingdom/buildings/0', () => {
-    it('should return 400 when trying to get a building with id 0', (done) => {
+  describe('GET /kingdom/buildings/5e0a1ceabadb94131493817a', () => {
```

Comment:
@GuJialu, he said that we only need to use the id given by the mongoose. and the mongoose id looks like this -> 5e0a1ceabadb94131493817a

Reply:
Because naturally mongodb uses string id. It is an unnecessary layer to convert the id to numbers. Furthermore, the plugin we found to convert the string id to number id only has 5000 downloads per week.